x-pack/plugins/cloud_security_posture/public/pages/rules/rules_container.tsx

```diff
@@ -93,6 +94,43 @@ export const RulesContainer = () => {
     params.benchmarkVersion
   );
 
+  const rulesStates = useCspGetRulesStates();
+  const arrayRulesStates = Object.values(rulesStates.data || {});
+  const filteredRulesStates: any[] = arrayRulesStates.filter(
+    (e: any) =>
+      e.benchmark_id === params.benchmarkId && e.benchmark_version === 'v' + params.benchmarkVersion
+  );
+
+  const allNewRulesItems: CspBenchmarkRulesWithStatus[] = useMemo(() => {
+    if (data === undefined) return [];
+    return data?.items.reduce((res: any[], obj1) => {
+      const rulesKey =
+        obj1.metadata.benchmark.id +
+        ';' +
+        obj1.metadata.benchmark.version +
+        ';' +
+        obj1.metadata.benchmark.rule_number;
+      const match = rulesStates?.data?.[rulesKey];
+      if (match) {
+        res.push({
+          ...obj1,
+          ...{ status: rulesStates?.data?.[rulesKey].muted === true ? 'muted' : 'unmuted' },
+        });
+      } else {
+        res.push({ ...obj1, ...{ status: 'unmuted' } });
+      }
+      return res;
+    }, []);
+  }, [data, rulesStates?.data]);
+
+  const filteredAllNewRulesItems: CspBenchmarkRulesWithStatus[] = useMemo(() => {
+    if (enabledDisabledItemsFilter === 'enabled')
+      return allNewRulesItems?.filter((e) => e?.status === 'muted');
+    else if (enabledDisabledItemsFilter === 'disabled')
+      return allNewRulesItems?.filter((e) => e?.status === 'unmuted');
+    else return allNewRulesItems;
+  }, [allNewRulesItems, enabledDisabledItemsFilter]);
```

Comment:
regarding my previous comment about converting the filter to be a boolean, it would make this function look like this:

```
const filteredRulesWithStatuses: CspBenchmarkRulesWithStatus[] = useMemo(() => {
  return isDisabledRulesFilterEnabled
    ? rulesWithStatuses?.filter(e => e?.status === 'unmuted')
    : rulesWithStatuses;
}, [allNewRulesItems, isDisabledRulesFilterEnabled]);
```

Reply:
correct me if I'm wrong but this will only filter between unmuted and muted rules yea ? what if we don't enable any of the filters ? (show both enabled and disabled users). currently we have 3 states for this filter (enabled to show enabled rules, disabled to show disabled rules and no-filter to show all rules)